### ISSUE TITLE
Feature Request: Auto-Populate Find Field for Relinking

### ISSUE THREAD

**Issue by tangierc** [latenitefilms/BRAWToolbox#141]
A feature where the path of the selected clip in the relinking toolbox would auto populate in the Find field would be very helpful and cut down on some of the manual process.

**Comment by latenitefilms** [comment]
It does pre-populate the **Find** field when you first open the Toolbox. Are you saying you want it to update the **Find** field each time you change the selected clip in the Relink list? Another option could be a **Use Selected Path** button next to **Paste**?

**Comment by tangierc** [comment]
Yeah, that would be great. It only prepulated the first time.  I found my self having to scroll through the clips that have different paths, manually type in the path in the Find filed then make adjustments in the Replace field. A Use Selected Path button next to paste would help.  Anything to alleviate a lot of manual copy pasting of the Find and Replace fields or manual entry. Often the directory differences are just drive or folder (s) that can be pasted repeatedly in the Replace field as well.

**Comment by tangierc** [comment]
Actually it'd be even better if the relink toolbox can be more automated by just pointing to a drive or folder and it recursively drills down to find the correct files similar to the way FCP works if that's possible.

**Comment by latenitefilms** [comment]
This is possible, but tricky and complex. One of the reason it's tricky is that Apple breaks its own App Store rules, and Final Cut Pro isn't actually sandboxed, so it can access any files without any kind of user initiated permission. BRAW Toolbox however is sandboxed, so the user must specifically allow access to a files and folders.

So whilst it's possible we could add some ability to "scan" an entire drive that you grant sandbox access to, it's still tricky to code, especially when this feature would be used by very few, as generally speaking you always keep your rushes in the same folder structure, so it should just be a matter of finding and replacing your volume name to relink?

**Comment by latenitefilms** [comment]
I'm going to close this for now, as it's now solved by the Media Extension.

**Comment by tangierc** [comment]
Sure thing Chris. I tested it this evening. It was surreal to be honest. I was almost giddy that I was actually using Braw this way in FCP.

Tangier
